File: mayonez-base/src/main/java/mayonez/application/JKeyManager.java
package mayonez.application;

import mayonez.event.*;
import mayonez.graphics.*;
import mayonez.input.*;
import mayonez.input.events.*;

import java.awt.event.*;

/**
 * Receives keyboard input events from AWT.
 *
 * @author SlavSquatSuperstar
 */
@UsesEngine(EngineType.AWT)
class JKeyManager extends KeyAdapter implements KeyInputHandler {

    // Key Callbacks

    @Override
    public void keyPressed(KeyEvent e) {
        getEventSystem().broadcast(new KeyInputEvent(e.getKeyCode(), true));
    }

    @Override
    public void keyReleased(KeyEvent e) {
        getEventSystem().broadcast(new KeyInputEvent(e.getKeyCode(), false));
    }

    // Input Handler Overrides

    @Override
    public EventSystem<KeyInputEvent> getEventSystem() {
        return InputEvents.KEYBOARD_EVENTS;
    }

    @Override
    public int getKeyCode(Key key) {
        return key.getAwtCode();
    }

}

File: mayonez-base/src/main/kotlin/mayonez/input/keyboard/JKeyManager.kt
package mayonez.input.keyboard

import mayonez.event.*
import mayonez.graphics.*
import mayonez.input.*
import mayonez.input.events.*
import java.awt.event.*

/**
 * Receives keyboard input events from AWT.
 *
 * @author SlavSquatSuperstar
 */
@UsesEngine(EngineType.AWT)
internal class JKeyManager : KeyAdapter(), KeyInputHandler {

    // Key Callbacks

    override fun keyPressed(e: KeyEvent) {
        eventSystem.broadcast(KeyInputEvent(e.keyCode, true))
    }

    override fun keyReleased(e: KeyEvent) {
        eventSystem.broadcast(KeyInputEvent(e.keyCode, false))
    }

    // Event Handler Overrides

    override fun getEventSystem(): EventSystem<KeyInputEvent> {
        return InputEvents.KEYBOARD_EVENTS
    }

    override fun getKeyCode(key: Key): Int = key.awtCode

}